File: JAVA/KuantumKaosYoneticisiJava/src/models/AntiMadde.java
package models;

import interfaces.IKritik;

public class AntiMadde extends KuantumNesnesi implements IKritik {
    
    public AntiMadde(String id) {
        super(id, 10, 40); 
    }

    @Override
    public void AnalizEt() {
        azaltVeKontrolEt(25); // Stabilite 25 birim düşer.
        System.out.println("[AntiMadde - " + getID() + "] EVRENİN DOKUSU TİTRİYOR... Stabilite -25.");
    }

    @Override
    public void AcilDurumSogutmasi() {
        setStabilite(getStabilite() + 50);
        System.out.println("[AntiMadde - " + getID() + "] KRİTİK SOĞUTMA! Stabilite +50. Başarılı.");
    }
}

File: JAVASCRIPT/kuantum_kaos_yoneticisi_js/models/AntiMadde.js
// models/AntiMadde.js
const KuantumNesnesi = require('./KuantumNesnesi');

class AntiMadde extends KuantumNesnesi {
    constructor(id) {
        super(id, 10, 40); 
    }

    AnalizEt() {
        this._azaltVeKontrolEt(25);
        console.log(`[AntiMadde - ${this.ID}] EVRENİN DOKUSU TİTRİYOR... Stabilite -25.`);
    }

    // IKritik arayüz metodu
    AcilDurumSogutmasi() {
        this.stabilite += 50;
        console.log(`[AntiMadde - ${this.ID}] KRİTİK SOĞUTMA! Stabilite +50. Başarılı.`);
    }
}
module.exports = AntiMadde;